components/library/libcidr/Makefile

```diff
@@ -0,0 +1,111 @@
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2016 Jim Klimov
+#
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=	libcidr
+COMPONENT_VERSION=	1.2.3
+COMPONENT_SUMMARY=	LibCIDR is a library to manipulate CIDR IP addresses
+COMPONENT_SRC=	$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.xz
+COMPONENT_ARCHIVE_HASH=	\
+  sha256:afbe266a9839775a21091b0e44daaf890a46ea4c2d3f5126b3048d82b9bfbbc4
+COMPONENT_ARCHIVE_URL=	\
+  https://www.over-yonder.net/~fullermd/projects/$(COMPONENT_NAME)/$(COMPONENT_ARCHIVE)
+COMPONENT_PROJECT_URL=	https://www.over-yonder.net/~fullermd/projects/$(COMPONENT_NAME)
+COMPONENT_FMRI=	library/$(COMPONENT_NAME)
+COMPONENT_CLASSIFICATION=	System/Libraries
+COMPONENT_LICENSE=	BSD
+COMPONENT_LICENSE_FILE=	LICENSE
+
+include $(WS_TOP)/make-rules/prep.mk
+include $(WS_TOP)/make-rules/justmake.mk
+include $(WS_TOP)/make-rules/ips.mk
+
+CXXFLAGS.32=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
+CFLAGS.32=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
+
+CXXFLAGS+= $(CXXFLAGS.$(BITS))
+CFLAGS+= $(CFLAGS.$(BITS))
+CFLAGS+= -I$(SOURCE_DIR)/include
+CFLAGS+= -I$(SOURCE_DIR)/src/include
+
+COMPONENT_CUSTOM_ARGS += BITS="$(BITS)"
+COMPONENT_CUSTOM_ARGS += CC="$(CC)"
+COMPONENT_CUSTOM_ARGS += CXX="$(CXX)"
+COMPONENT_CUSTOM_ARGS += CPP="$(CPP)"
+COMPONENT_CUSTOM_ARGS += LD="$(LD)"
+COMPONENT_CUSTOM_ARGS += LN="$(LN)"
+COMPONENT_CUSTOM_ARGS += RM="$(RM)"
+COMPONENT_CUSTOM_ARGS += MV="$(MV)"
+COMPONENT_CUSTOM_ARGS += MKDIR="$(MKDIR)"
+COMPONENT_CUSTOM_ARGS += SED="$(GSED)"
+COMPONENT_CUSTOM_ARGS += INSTALL="$(INSTALL)"
+COMPONENT_CUSTOM_ARGS += CFLAGS="$(CFLAGS)"
+COMPONENT_CUSTOM_ARGS += CPPFLAGS="$(CPPFLAGS)"
+COMPONENT_CUSTOM_ARGS += CXXFLAGS="$(CXXFLAGS)"
+COMPONENT_CUSTOM_ARGS += LDFLAGS="$(LDFLAGS)"
+
+USRLIBDIR.32 = $(USRLIBDIR)
+USRLIBDIR.64 = $(USRLIBDIR64)
+USRBINDIR.32 = $(USRBINDIR)
+USRBINDIR.64 = $(USRBINDIR64)
+
+COMPONENT_CUSTOM_ARGS += PREFIX="$(USRDIR)"
+COMPONENT_CUSTOM_ARGS += CIDR_LIBDIR="$(USRLIBDIR.$(BITS))"
+COMPONENT_CUSTOM_ARGS += CIDR_BINDIR="$(USRBINDIR.$(BITS))"
+COMPONENT_CUSTOM_ARGS += CIDR_INCDIR="$(USRINCDIR)"
+COMPONENT_CUSTOM_ARGS += CIDR_MANDIR="$(USRSHAREMANDIR)"
+COMPONENT_CUSTOM_ARGS += CIDR_DOCDIR="$(USRSHAREDIR)/$(COMPONENT_NAME)/docs"
+COMPONENT_CUSTOM_ARGS += CIDR_EXDIR="$(USRSHAREDIR)/$(COMPONENT_NAME)/examples"
+
+COMPONENT_BUILD_ARGS += $(COMPONENT_CUSTOM_ARGS)
+COMPONENT_INSTALL_ARGS += $(COMPONENT_CUSTOM_ARGS)
+
+COMPONENT_PREP_ACTION = \
+    ( cd $(@D) && \
+      $(GSED) -e 's|\-Wl\,\-x|-m\$$\{BITS\}|' \
+        -e 's|^\([ \t]*L_FLAGS.*=[ \t]*\)|\1\$$\{LDFLAGS\} |' \
+        -i src/Makefile.inc && \
+      $(GSED) -e 's|\(^[ \t]*\)awk |\1\$(NAWK) |g' \
+        -e 's|echo -n \(\".*\"\)|/bin/echo \1"\\c"|' \
+        -i mkgmake.sh )
+
+COMPONENT_PRE_BUILD_ACTION = \
+    ( $(CLONEY) $(SOURCE_DIR) $(@D) && \
+      cd $(@D) && \
+      ./rmgmake.sh && ./mkgmake.sh )
+
+COMPONENT_POST_INSTALL_ACTION = \
```

Comment:
I think it is a poor idea - there will be more manual work to update them
consistently as we re-spin the package. Currently they are generated with
ARCH-dependent paths, proper versions, etc...

2016-03-30 14:31 GMT+02:00 Alexander Pyhalov <notifications@github.com>:

> In components/library/libcidr/Makefile
> <https://github.com/OpenIndiana/oi-userland/pull/1812#discussion_r57879592>
> :
>
> > +
> > +COMPONENT_PREP_ACTION = \
> > +    ( cd $(@D) && \
> > +      $(GSED) -e 's|\-Wl\,\-x|-m\$$\{BITS\}|' \
> > +        -e 's|^\([ \t]*L_FLAGS.*=[ \t]*\)|\1\$$\{LDFLAGS\} |' \
> > +        -i src/Makefile.inc && \
> > +      $(GSED) -e 's|\(^[ \t]*\)awk |\1\$(NAWK) |g' \
> > +        -e 's|echo -n \(\".*\"\)|/bin/echo \1"\\c"|' \
> > +        -i mkgmake.sh )
> > +
> > +COMPONENT_PRE_BUILD_ACTION = \
> > +    ( $(CLONEY) $(SOURCE_DIR) $(@D) && \
> > +      cd $(@D) && \
> > +      ./rmgmake.sh && ./mkgmake.sh )
> > +
> > +COMPONENT_POST_INSTALL_ACTION = \
>
> Please, deliver *.pc files as files in files/ subdirectory.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenIndiana/oi-userland/pull/1812/files/dd86909c789b9dc971cc57430cffceb67d6cc9b2#r57879592>
>